### PR TITLE
Convert booleans arguments in build_arg_string, not in kwargs_to_strings

### LINKED ISSUE #640: Decorator kwargs_to_strings is too aggressive when converting boolean arguments
**Description of the problem**

https://github.com/GenericMappingTools/pygmt/blob/c191d3e09a43a0536b2ae14a7c8380969f4646a4/pygmt/helpers/decorators.py#L279

Decorator `kwargs_to_strings` can deal with boolean arguments. For example, dict `{'A': True, 'B': False, 'R': '0/10/0/10'}` will be converted to `{'A': '', 'R': '0/10/0/10'}` first, and then `build_arg_strings` will further convert it to valid GMT arguments `-A -R0/10/0/10`. 

The decorator works well for arguments that will be passed to GMT, but doesn't work for pure Python arguments.

**Full code that generated the error**

```python
from pprint import pprint
from pygmt.helpers.decorators import kwargs_to_strings

@kwargs_to_strings(R='sequence', files='sequence_space')
def module(nongmt_args=True, *args, **kwargs):
    "A module that prints the arguments it received"
    pprint(args)
    pprint(kwargs)
    if nongmt_args is True:
        print("nongmt_args is still True. Do something in PyGMT.")
    elif nongmt_args == "":
        print("the kwargs_to_strings decorator converts nongmt_args to an empty string.")
    else:
        print(f"nongmt_args is '{nongmt_args}'")

module(R=[1, 2, 3, 4], nongmt_args=True)
```

**Output**
```
()
{'R': '1/2/3/4'}
the kwargs_to_strings decorator converts nongmt_args to an empty string.
```
As shown in the output above, the boolean argument `nongmt_args` is converted from `True` to an empty string `""`. Thus, checking `nongmt_args is True` no longer works.

If removing the `kwargs_to_strings` decorator before the `module` function, the output is:
```
()
{'R': [1, 2, 3, 4]}
nongmt_args is still True. Do something in PyGMT.
```

**Possible solution**

Like what we do in #639, the possible solution is to convert boolean arguments in `build_arg_strings` function before passing arguments to GMT, not in the `kwargs_to_strings` decorator.

**System information**

Please paste the output of `python -c "import pygmt; pygmt.show_versions()"`:

```
PyGMT information:
  version: v0.2.0+19.gc191d3e0
System information:
  python: 3.8.3 (default, Jul  2 2020, 11:26:31)  [Clang 10.0.0 ]
  executable: /Users/seisman/.anaconda/bin/python
  machine: macOS-10.15.6-x86_64-i386-64bit
Dependency information:
  numpy: 1.18.5
  pandas: 1.0.5
  xarray: 0.16.1
  netCDF4: 1.5.3
  packaging: 20.4
  ghostscript: 9.53.1
  gmt: 6.2.0_612863b_2020.10.03
GMT library information:
  binary dir: /Users/seisman/.anaconda/bin
  cores: 8
  grid layout: rows
  library path: /Users/seisman/local/GMT/lib/libgmt.dylib
  padding: 2
  plugin dir: /Users/seisman/local/GMT/lib/gmt/plugins
  share dir: /Users/seisman/local/GMT/share
  version: 6.2.0
```

### PR DESCRIPTION
**Pull request by seisman**
**Description of proposed changes**

See #640 and #639 (the discussions and the fix) for context.

Changes in this PR:

- 06bee27e4dd3e57b3d37078feb2e97b771cb057c: Remove `convert_bools` parameter from `kwargs_to_strings`. So `kwargs_to_strings` won't convert boolean arguments.
- b7210ff3d855903b5277b3cada3964f57120af4a: Handle boolean arguments in the `build_arg_string` function. 
- 59a2b281bdba1f72dd3ceb83611b58bea7a88679: The boolean `use_srtm` parameter of `load_earth_relief` won't be converted to an empty string or removed anymore. Now it's safe to check `if use_strm`.
- a9d167d9ab35a249660e58cfdade2eb88fd930a3: The `autolabel` parameter of `subplot` function can be either bool or str, but we had to check if `autolabel is not None`. It's not intuitive. Actually, we made a mistake in the initial version (4126c16160c67f8d2c405e7d5da56145adc989d4) and then fixed it (eadb8470065da5436802bc796e868e1ac28d8ac7).
- aac61c25a404b17665a227f362c99d11e6bf2b26: Similarly, `justify`, `angle`, and `font` can be boolean, but we have to check if it's `None`. This commit also fixes the weird codes.

Fixes #640.

**Reminders**

- [x] Run `make format` and `make check` to make sure the code follows the style guide.
- [x] Add tests for new features or tests that would have caught the bug that you're fixing.
- [ ] Add new public functions/methods/classes to `doc/api/index.rst`.
- [ ] Write detailed docstrings for all functions/methods.
- [ ] If adding new functionality, add an example to docstrings or tutorials.

**Slash Commands**

You can write slash commands (`/command`) in the first line of a comment to perform
specific operations. Supported slash commands are:

- `/format`: automatically format and lint the code
- `/test-gmt-dev`: run full tests on the latest GMT development version

### CODE REVIEW

File: pygmt/helpers/utils.py [hunk 1]
```diff
@@ -128,10 +129,20 @@ def build_arg_string(kwargs):
...
+    ...         dict(
+    ...             R="1/2/3/4",
+    ...             J="X4i",
+    ...             P="",
+    ...             E=200,
+    ...             X=None,
+    ...             Y=None,
+    ...             A=True,
+    ...             B=False,
+    ...             Z=0,
```
Comment (weiji14) [inline pygmt/helpers/utils.py:141]: Sort alphabetically and use SI units :slightly_smiling_face:

```suggestion
    ...             A=True,
    ...             B=False,
    ...             E=200,
    ...             J="X4c",
    ...             P="",
    ...             R="1/2/3/4",
    ...             X=None,
    ...             Y=None,
    ...             Z=0,
```

File: pygmt/helpers/utils.py [hunk 2]
```diff
@@ -128,10 +129,20 @@ def build_arg_string(kwargs):
...
+    ...         )
     ...     )
     ... )
-    -E200 -JX4i -P -R1/2/3/4
+    -A -E200 -JX4i -P -R1/2/3/4 -Z0
```
Comment (weiji14) [inline pygmt/helpers/utils.py:145]: Use SI units.
```suggestion
    -A -E200 -JX4c -P -R1/2/3/4 -Z0
```

**Review by weiji14 — COMMENTED** [review]
Just some minor comments for now. I think there's some room for improvement but will suggest more later (possibly after the weekend).

File: pygmt/src/text.py [hunk 3]
```diff
@@ -177,13 +177,23 @@ def text_(
         )
     ):
         kwargs.update({"F": ""})
-    if angle is not None and isinstance(angle, (int, float, str)):
+
+    if angle is True:
+        kwargs["F"] += "+a"
+    elif isinstance(angle, (int, float, str)):
         kwargs["F"] += f"+a{str(angle)}"
```
Comment (weiji14) [inline pygmt/src/text.py:184]: Just thinking ahead to resolve #483 (allowing list inputs to angle/list/justify), we could simply use `if angle` (returns True for list/int/float/str inputs) like so:

```suggestion
    if angle:
        kwargs["F"] += "+a"
        if isinstance(angle, (int, float, str)):
            kwargs["F"] += str(angle)
```

File: pygmt/src/text.py [hunk 4]
```diff
@@ -177,13 +177,23 @@ def text_(
...
-    if font is not None and isinstance(font, str):
+
+    if font is True:
+        kwargs["F"] += "+f"
+    elif isinstance(font, str):
         kwargs["F"] += f"+f{font}"
```
Comment (weiji14) [inline pygmt/src/text.py:189]:
```suggestion
    if font:
        kwargs["F"] += "+f"
        if isinstance(font, str):
            kwargs["F"] += font
```

File: pygmt/src/text.py [hunk 5]
```diff
@@ -177,13 +177,23 @@ def text_(
...
-    if justify is not None and isinstance(justify, str):
+
+    if justify is True:
+        kwargs["F"] += "+j"
+    elif isinstance(justify, str):
         kwargs["F"] += f"+j{justify}"
```
Comment (weiji14) [inline pygmt/src/text.py:194]:
```suggestion
    if justify:
        kwargs["F"] += "+j"
        if isinstance(justify, str):
            kwargs["F"] += justify
```

**Review by weiji14 — COMMENTED** [review]
A few suggestions, mostly on refactoring the if-then blocks. Feel free to disagree if it feels less readable to you.

File: pygmt/src/text.py (hunk 3 above)
Comment (seisman) [reply pygmt/src/text.py:184]: Actually, I was planing to work on #483, but I realized that the `Figure.text()` function needs more refactoring, so I opened PR #1121.

Personally, I prefer to make the changes to `text.py` as small as possible, and will address #483 after #1121 is merged.

Comment (weiji14) [reply pygmt/src/text.py:184]: Ok, as you wish then.

**Review by weiji14 — APPROVED** [review]
Thanks @seisman, good to finally close #640, and this change will open up a lot of new possibilities with PyGMT functions!

**Pull request merged**